src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeBlockTagsGroupCheck.java

```diff
@@ -0,0 +1,243 @@
+        // No need to filter token because overridden getDefaultJavadocTokens ensures that we only
+        // receive JAVADOC_TAG DetailNode.
+        if (!isAnotherTagBefore(tagNode)
+                && !isOnlyTagInWholeJavadoc(tagNode)
+                && insufficientConsecutiveNewlines(tagNode)) {
+            log(tagNode.getLineNumber(),
+                    MSG_JAVADOC_TAG_LINE_BEFORE,
+                    tagNode.getChildren()[0].getText());
+        }
+    }
+
+    private static boolean isAnotherTagBefore(DetailNode tagNode) {
+        boolean found = false;
+        DetailNode currentNode = JavadocUtil.getPreviousSibling(tagNode);
+        while (currentNode != null) {
+            if (currentNode.getType() == JavadocTokenTypes.JAVADOC_TAG) {
+                found = true;
+                break;
+            }
+            currentNode = JavadocUtil.getPreviousSibling(currentNode);
+        }
+        return found;
+    }
+
+    /**
+     * When javadoc has only a javadoc tag like {@literal @} in it, the JAVADOC_TAG in a JAVADOC
+     * detail node will always have 2 or 3 siblings before it. The parse tree looks like:
+     * <pre>
+     * JAVADOC[3x0]
+     * |--NEWLINE[3x0] : [\n]
+     * |--LEADING_ASTERISK[4x0] : [ *]
+     * |--WS[4x2] : [ ]
+     * |--JAVADOC_TAG[4x3] : [@param T The bar.\n ]
+     * </pre>
+     * Or it can also look like:
+     * <pre>
+     * JAVADOC[3x0]
+     * |--NEWLINE[3x0] : [\n]
+     * |--LEADING_ASTERISK[4x0] : [ *]
+     * |--JAVADOC_TAG[4x3] : [@param T The bar.\n ]
+     * </pre>
+     * We do not include the variation
+     * <pre>
+     *  /**&#64;param noSpace there is no space here
+     * </pre>
+     * which results in the tree
+     * <pre>
+     * JAVADOC[3x0]
+     * |--JAVADOC_TAG[4x3] : [@param noSpace there is no space here\n ]
+     * </pre>
+     * because this one is invalid. We must recommend placing a blank line to separate &#64;param
+     * from the first javadoc asterisks.
+     *
+     * @param tagNode the at tag node to check if there is nothing before it.
+     * @return true if there no text before the tagNode.
+     */
+    private static boolean isOnlyTagInWholeJavadoc(DetailNode tagNode) {
+        final List<Integer> previousNodeTypes = new ArrayList<>();
+        DetailNode currentNode = JavadocUtil.getPreviousSibling(tagNode);
+        while (currentNode != null) {
+            previousNodeTypes.add(currentNode.getType());
+            currentNode = JavadocUtil.getPreviousSibling(currentNode);
+        }
+        return ONLY_TAG_VARIATION_1.equals(previousNodeTypes)
+                || ONLY_TAG_VARIATION_2.equals(previousNodeTypes);
+    }
+
+    /**
+     * Iterates through the previous siblings of the tagNode looking for empty lines until
+     * there are no more siblings or it hits something other than asterisk, whitespace or newline.
+     * If it finds at least one empty line, return true. Return false otherwise.
```

Comment:
First sentence of the javadoc is the most important and says what it is doing for the user calling it. Javadoc printers will only show it for summaries. First sentence is pretty long and 2nd and 3rd still say what its doing. Can we redo the first sentence so it gives a clear informaition and leave further sentences to be implementation specific?